lib/sqlalchemy/dialects/mysql/mariadb.py

```diff
@@ -7,13 +7,23 @@
 # mypy: ignore-errors
 from .base import MariaDBIdentifierPreparer
 from .base import MySQLDialect
+from .base import MySQLTypeCompiler
+
+
+class MariaDBTypeCompiler(MySQLTypeCompiler):
+    def visit_uuid(self, type_, **kw):
+        if self.dialect.server_version_info >= (10, 7) and type_.native_uuid:
```

Comment:
**Michael Bayer** (zzzeek) wrote:

sorry, I know I said the code was "fine" in my previous comment, but I forgot about this.

Please write this as:


    if (
        self.dialect.server_version_info is not None
        and self.dialect.server_version_info < (10, 7)
    ) or not type_.native_uuid:
        return super().visit_uuid(type_, **kw)
    else:
        return self.visit_UUID(type_, **kw)

View this in Gerrit at https://gerrit.sqlalchemy.org/c/sqlalchemy/sqlalchemy/+/5120

Reply:
thanks, updated